### PR TITLE
fix(ui) : Shop detail panel click scrolling added

### PR DESCRIPTION
**Pull request by flowers9**
**Bugfix:**

## The Bug
On the Shipyard/Outfitter screens, the detail panel (the one in the middle, that has the details of whatever's currently selected in the left panel) has scroll buttons that can't be clicked (although the panel is scrollable other ways, by keypress or scroll mouse).

## Fix Details
Added a section to ShopPanel::Click() to check for clicks on those buttons.

## Testing Done
Opened a game, selected a ship with enough details to be scrollable, tried clicking the buttons.

### CODE REVIEW

**Review by Hurleveur — APPROVED** [review]
I can confirm the bug and this seems like a good fix that also clarifies where each set of arrows are. Looks good code wise.

File: source/ShopPanel.cpp [hunk 1]
```diff
@@ -760,14 +760,23 @@ bool ShopPanel::Click(int x, int y, int /* clicks */)
 	if(button)
 		return DoKey(button);
 
-	// Check for clicks in the scroll arrows.
+	// Check for clicks in the ShipsSidebar Panel arrows.
```
Comment (warp-core) [inline source/ShopPanel.cpp:763]:
```suggestion
	// Check for clicks on the ShipsSidebar pane arrows.
```

File: source/ShopPanel.cpp [hunk 2]
```diff
@@ -760,14 +760,23 @@ bool ShopPanel::Click(int x, int y, int /* clicks */)
...
 	if(x >= Screen::Right() - 20)
 	{
 		if(y < Screen::Top() + 20)
 			return Scroll(0, 4);
 		if(y < Screen::Bottom() - BUTTON_HEIGHT && y >= Screen::Bottom() - BUTTON_HEIGHT - 20)
 			return Scroll(0, -4);
 	}
+	// Check for clicks in the DetailsSidebar Panel arrows.
```
Comment (warp-core) [inline source/ShopPanel.cpp:771]:
```suggestion
	// Check for clicks on the DetailsSidebar pane arrows.
```
They aren't really their own panels.

File: source/ShopPanel.cpp [hunk 3]
```diff
@@ -760,14 +760,23 @@ bool ShopPanel::Click(int x, int y, int /* clicks */)
...
+	else if(x >= Screen::Right() - SIDEBAR_WIDTH - 20 && x < Screen::Right() - SIDEBAR_WIDTH)
+	{
+		if(y < Screen::Top() + 20)
+			return Scroll(0, 4);
+		if(y >= Screen::Bottom() - 20)
+			return Scroll(0, -4);
+	}
+	// Check for clicks in the Main Panel arrows.
```
Comment (warp-core) [inline source/ShopPanel.cpp:779]:
```suggestion
	// Check for clicks on the Main pane arrows.
```

**Pull request merged**